File: app/src/main/java/danb/speedrunbrowser/utils/PreCachingGridLayoutManager.java
package danb.speedrunbrowser.utils;

import android.content.Context;
import android.util.AttributeSet;

import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class PreCachingGridLayoutManager extends GridLayoutManager {
    private static final int DEFAULT_EXTRA_LAYOUT_SPACE = 600;
    private int extraLayoutSpace = -1;
    private Context context;

    public PreCachingGridLayoutManager(Context context, AttributeSet attrs, int spanCount, int t2) {
        super(context, attrs, spanCount, t2);
        this.context = context;
        this.extraLayoutSpace = DEFAULT_EXTRA_LAYOUT_SPACE;
    }

    public void setExtraLayoutSpace(int extraLayoutSpace) {
        this.extraLayoutSpace = extraLayoutSpace;
    }

    @Override
    protected int getExtraLayoutSpace(RecyclerView.State state) {
        if (extraLayoutSpace > 0) {
            return extraLayoutSpace;
        }
        return DEFAULT_EXTRA_LAYOUT_SPACE;
    }
}

File: app/src/main/java/danb/speedrunbrowser/utils/PreCachingGridLayoutManager.kt
package danb.speedrunbrowser.utils

import android.content.Context
import android.util.AttributeSet

import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView

const val DEFAULT_EXTRA_LAYOUT_SPACE = 600

class PreCachingGridLayoutManager(context: Context, attrs: AttributeSet, spanCount: Int, t2: Int) : GridLayoutManager(context, attrs, spanCount, t2) {
    var extraLayoutSpace = DEFAULT_EXTRA_LAYOUT_SPACE

    override fun getExtraLayoutSpace(state: RecyclerView.State): Int {
        return if (extraLayoutSpace > 0) {
            extraLayoutSpace
        } else DEFAULT_EXTRA_LAYOUT_SPACE
    }
}